recurring_payments_stripe/models/account_move.py

```diff
@@ -0,0 +1,169 @@
+                        Payment = self.env["account.payment"].sudo()
+                        payment_vals = {
+                            "journal_id": self.env["account.journal"]
+                            .search([("type", "=", "bank")], limit=1)
+                            .id,
+                            "amount": invoice.amount_total,
+                            "payment_type": "inbound",
+                            "partner_type": "customer",
+                            "partner_id": invoice.partner_id.id,
+                            "payment_method_id": self.env.ref(
+                                "account.account_payment_method_manual_in"
+                            ).id,
+                            "ref": f"Stripe PaymentIntent {payment_intent['id']}",
+                        }
+                        payment = Payment.create(payment_vals)
+                        payment.action_post()
+                        invoice.payment_state = "paid"
+                    elif payment_intent["status"] == "requires_action":
+                        raise UserError(
+                            "El pago requiere autenticación adicional (3D Secure)."
+                        )
+                    else:
+                        raise UserError(
+                            f"Error en el pago de Stripe: {payment_intent['status']}"
+                        )
+
+                except stripe.StripeError as e:
+                    raise UserError(f"Error de Stripe: {e.user_message or str(e)}")
+
+            else:
+                # Si no es una suscripción recurrente o no tiene método de pago en Stripe, llama al método original
+                return super(AccountMove, self).action_register_payment()
+
+    @api.model
+    def cron_process_due_invoices(self):
+        """Procesa el pago de facturas vencidas para suscripciones recurrentes."""
+        # Obtener la fecha de hoy
+        today = date.today()
+
+        # Buscar facturas de suscripciones recurrentes que vencen hoy y están abiertas
+        invoices = self.search(
+            [
+                ("state", "=", "posted"),
+                ("payment_state", "=", "not_paid"),
+            ]
+        )
+
+        for invoice in invoices:
+            # Buscar la suscripción asociada a la factura
+            subscription = self.env["sale.subscription"].search(
+                [("code", "=", invoice.invoice_origin)], limit=1
+            )
+
+            # Verificar si es una suscripción recurrente con Stripe
+            if subscription and subscription.is_recurrent:
+                try:
+                    # Intentar registrar el pago en la factura
+                    invoice.action_register_payment()
+                except Exception as e:
+                    # Capturar excepciones en caso de error en el registro de pago
+                    _logger.error(
+                        f"Error al procesar el pago de la factura {invoice.id}: {str(e)}"
+                    )
```

Comment:
I think this can be more efficient:
```suggestion
        invoices = self.search(
            [
                ("state", "=", "posted"),
                ("payment_state", "in", ["not_paid","partial]),
                ("subscription_id", "!=", False),
                ("subscription_id.charge_automatically", "=", True),
            ]
        )

        for invoice in invoices:
            try:
                # Try to process the payment on the invoice
                invoice.action_register_payment()
            except Exception as e:
                # Capture exceptions in case of error in payment 
                _logger.error(
                    f"Error processing payment for invoice  {invoice.id}: {str(e)}"
                )
```